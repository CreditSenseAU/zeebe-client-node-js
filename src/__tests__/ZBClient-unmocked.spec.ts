import { ZBClient } from '..'

jest.setTimeout(13000)

describe('ZBClient constructor', () => {
	it('throws an exception when there is no broker and retry is false', async () => {
		const zbc = new ZBClient('localhoster', { retry: false })
		expect.assertions(1)
		try {
			await zbc.deployWorkflow(
				'./src/__tests__/testdata/hello-world.bpmn'
			)
		} catch (e) {
			expect(e.message.indexOf('14 UNAVAILABLE:')).toEqual(0)
		}
	})
	it('does not throw when there is no broker, by default', async done => {
		const zbc = new ZBClient('localhoster')
		setTimeout(async () => {
			// tslint:disable-next-line
			console.log(
				'^^^ The gRPC connection failure message above is expected. ^^^'
			)
			await zbc.close()
			expect(true).toBe(true)
			// We have to wait ten seconds here, because the operation retry logic keeps it alive
			setTimeout(() => done(), 10000)
		}, 2000)
		// tslint:disable-next-line
		console.log(
			'vvv The gRPC connection failure message below is expected. vvv'
		)
		await zbc.deployWorkflow('./src/__tests__/testdata/hello-world.bpmn')
	})
	it('throws an exception when workflowInstanceKey is malformed', async () => {
		const zbc = new ZBClient('localhoster', { retry: false })
		expect.assertions(1)
		try {
			await zbc.cancelWorkflowInstance('hello-world')
		} catch (e) {
			expect(e).toMatchSnapshot()
		}
	})
})
