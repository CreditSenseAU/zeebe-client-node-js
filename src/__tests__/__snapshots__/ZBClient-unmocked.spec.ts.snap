// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cancelWorkflowInstance throws an exception when workflowInstanceKey is malformed 1`] = `
[Error: 
		  workflowInstanceKey is malformed, value : hello-world
		  ]
`;
